### PR TITLE
Add --test-user flag to integration test, default to jenkins if the run is triggered from docker image

### PR DESCRIPTION
**Pull request by krzyzacy**
To fix the integration test under new image

### CODE REVIEW

**Comment by krzyzacy** [comment]
I definitely did not change historical_handlers_test.go

**Comment by krzyzacy** [comment]
cc @DirectXMan12 

I have a green run when hard-code jenkins@ into the code, now add it properly as a flag.

File: integration/heapster_api_test.go [hunk 1]
```diff
@@ -174,7 +175,11 @@ func buildAndPushDockerImages(fm kubeFramework, zone string) error {
 	}
 	hostnames := []string{}
 	for _, node := range nodes {
-		hostnames = append(hostnames, strings.Split(node, ".")[0])
+		nodeHost := strings.Split(node, ".")[0]
+		if *testUser != "" {
+			nodeHost = fmt.Sprintf("jenkins@%s", nodeHost)
```
Comment (DirectXMan12) [inline integration/heapster_api_test.go]:
```go
if *testUser != "" {
    nodeHost = fmt.Sprintf("%s@%s", *testUser, nodeHost)
}
```

Comment (krzyzacy) [inline integration/heapster_api_test.go]: oh......whoops

Comment (krzyzacy) [inline integration/heapster_api_test.go]: how did e2e pass

Comment (krzyzacy) [inline integration/heapster_api_test.go]: puh I see

**Comment by krzyzacy** [comment]
Aha `Jenkins overloaded. Please try again later.` everywhere... will retrigger it once we drain the queue...

File: integration/heapster_api_test.go (hunk 1 above)
Comment (krzyzacy) [inline integration/heapster_api_test.go]: Should it be reference or not &_@ I wrote too much python and forgot how go works...

**Comment by krzyzacy** [comment]
@k8s-bot test this

**Comment by krzyzacy** [comment]
greened, thanks Jenkins!

**Pull request merged**